packages/router/history.ts

```diff
@@ -282,22 +285,24 @@ export function createMemoryHistory(
       index += 1;
       entries.splice(index, entries.length, nextLocation);
       if (v5Compat && listener) {
-        listener({ action, location: nextLocation });
+        listener({ action, location: nextLocation, delta: 1 });
```

Comment:
do we want deltas on push/replace?

Reply:
They're not really useful or harmful, this was mostly a hack to get some of our snapshot/object tests to pass. I'll investigate a proper fix for those later.

Reply:
I'm not sure if we'd need them for `<BrowserRouter>` but they could be dropped for `RouterProvider` for now since we never hit this code path?